### DISCUSSION TITLE
Packaging to Java behind a corporate proxy.

### DISCUSSION THREAD

**Discussion by Marty-Me** [aws/jsii#3465] [Q&A]
### Affected Languages

- [ ] `TypeScript` or `Javascript`
- [ ] `Python`
- [X] `Java`
- [ ] .NET (`C#`, `F#`, ...)

### General Information
* **JSII Version:** 1.16.0 (build 99a3413), typescript 3.9.7
* **Platform:**     
Darwin - 19.6.0 Darwin Kernel Version 19.6.0: Thu Oct 29 22:56:45 PDT 2020; root:xnu-6153.141.2.2~1/RELEASE_X86_64 x86_64
### The Question
I'm trying to run `npm run build` with with a `java` block in `targets`, which executes the following command in one of its steps:
`"mvn --batch-mode deploy -D=altDeploymentRepository=local::default::file:///var/folders/tq/kpf08kxn6pl0b92jn8rmr9xc0000gn/T/npm-packLPDcsR --settings=user.xml`

This does however override the settings in my ~/.m2/settings.xml file, which changes the default-repository to a maven-central-proxy provided for developers.

As a result I am unable to download the required artifacts:
`Could not transfer artifact org.apache.maven.plugins:maven-source-plugin:pom:3.2.1 from/to central (https://repo.maven.apache.org/maven2)`

Is there any way for me to configure jsii-pacmac to still use my custom artifact repository?

**Comment by Marty-Me** [comment]
Hi, I'm still facing this issue, does anyone have some guidance for me?

**Comment by mjh5153** [comment]
I am also experiencing this issue. Any updates?

typescript 3.9.7
jsii 1.53, mvn 3.2.1
Java 11.0.12
mvn 3.8.4
node 14.5

**Comment by icj217** [comment]
I have a PR up (https://github.com/aws/jsii/pull/3893) that should hopefully address this problem. Would appreciate any feedback on the PR to confirm the implementation makes sense and is compatible with your use case.

**Comment by tanguc** [comment]
[Solved here](https://github.com/aws/jsii/pull/4068)

`jsii-pacmak generate . --mvn-quiet= --mvn-threads=3`

make sure to add your argument(s) starting by `--mvn-`

tips;
> The former settings.xml are also called global settings, the latter settings.xml are referred to as user settings. If both files exists, their contents gets merged, with the user-specific settings.xml being dominant.
https://maven.apache.org/settings.html
